arithmetization/src/test/java/net/consensys/linea/replaytests/ReplayTests.java

```diff
@@ -188,4 +197,43 @@ void mainnet1339346ContextRevertTwice() {
   void legacyTxWithoutChainID() {
     replay(LINEA_SEPOLIA, "254251.sepolia.json.gz");
   }
+
+  static List<Arguments> blockNumbers = new ArrayList<>();
+
+  @ParameterizedTest
+  @MethodSource("replayBlockTestSource")
+  void replayBlockTest(int blockNumber) {
+    // TODO: fix issue with paths
+    File file = new File("arithmetization/src/test/resources/replays/" + blockNumber + ".json.gz");
+    if (!file.exists()) {
+      String command = "./scripts/capture.pl --start " + blockNumber;
+      try {
+        // Execute the command
+        Process process = Runtime.getRuntime().exec(command);
+        process.waitFor();
+      } catch (InterruptedException | IOException e) {
+        e.printStackTrace();
+      }
+    }
+    // replay(LINEA_MAINNET, blockNumber + ".json.gz");
+    System.out.println("Replaying block " + blockNumber);
+  }
+
+  static Stream<Arguments> replayBlockTestSource() {
+    // Example of how to add a range
+    add(2435888, 2435889);
+    // Example of how to add a single block
+    add(2435890);
+    return blockNumbers.stream();
+  }
+
+  private static void add(int start, int end) {
+    for (int i = start; i <= end; i++) {
+      blockNumbers.add(Arguments.of(i));
+    }
+  }
+
+  private static void add(int start) {
+    blockNumbers.add(Arguments.of(start));
```

Comment:
Have you looked at the python script ? It produces a `.csv` file which you can then download in bulk with @gusiri 's `bulk_capture` script.

Reply:
Yes the python one, but cannot find the `bulk_capture` one. However, since the purpose of this test is having one parametric test per block, I believe it makes sense to download them one by one and have separated files.